drivers/usb/device/usb_dc_dw.c

```diff
@@ -719,7 +719,12 @@ int usb_dc_attach(void)
 	IRQ_CONNECT(DT_INST_IRQN(0),
 		    DT_INST_IRQ(0, priority),
 		    usb_dw_isr_handler, 0,
+#ifdef CONFIG_GIC_V1
+		    DT_INST_IRQ(0, type));
+#else
 		    DT_INST_IRQ(0, sense));
+#endif
+
```

Comment:
Can you explain this change?

Reply:
Hi @carlocaione,

First of all the use of DT_INST_IRQ (0, sense), is very specific to snps,designware-intc... You can see this in the biding documentation  https://docs.zephyrproject.org/latest/build/dts/api/bindings/interrupt-controller/snps,designware-intc.html.... so, the USB driver shouldn't have that definition when using it with arm,gic interrupt controller.

In the modification, we just decided to configure the "type" cell in the binding. Instead of the "sense" cell... This does not mean that both of them are interchangeable.

I believe, we can delete "DT_INST_IRQ(0, type)" from the file, as is not strictly necessary. But the use of "DT_INST_IRQ(0, sense)" should be restricted only when using the Synopsys interrupt controller.

~E.V.

Reply:
uhm, this sucks. @mbolivar-nordic do you have any smart solution about this? The change is fine but it is definitely not scalable if we decide to use this IP with different interrupt controllers with different cell names.

Reply:
@mevalver So, based on your comment, the logic should be "if its interrupt parent is `snps,designware-intc,` use `sense`; else, use `type`."

Reply:
I think that could be the case, as "sense" is specific cell-name to snps,designware-intc